tests/tests/runner.rs

```diff
@@ -0,0 +1,299 @@
+        chain_head_update_listener,
+        Arc::new(StaticStreamBuilder { events }),
+        ethereum::ENV_VARS.reorg_threshold,
+        // We assume the tested chain is always ingestible for now
+        true,
+    );
+
+    let mut blockchain_map = BlockchainMap::new();
+    blockchain_map.insert(network_name.clone(), Arc::new(chain));
+
+    let static_filters = ENV_VARS.experimental_static_filters;
+
+    let blockchain_map = Arc::new(blockchain_map);
+    let subgraph_instance_manager = SubgraphInstanceManager::new(
+        &logger_factory,
+        subgraph_store.clone(),
+        blockchain_map.clone(),
+        mock_registry.clone(),
+        link_resolver.cheap_clone(),
+        static_filters,
+    );
+
+    // Create IPFS-based subgraph provider
+    let subgraph_provider: Arc<
+        IpfsSubgraphAssignmentProvider<
+            SubgraphInstanceManager<graph_store_postgres::SubgraphStore>,
+        >,
+    > = Arc::new(IpfsSubgraphAssignmentProvider::new(
+        &logger_factory,
+        link_resolver.cheap_clone(),
+        subgraph_instance_manager,
+    ));
+
+    let panicking_subscription_manager = Arc::new(PanicSubscriptionManager {});
+
+    let subgraph_registrar: Arc<
+        IpfsSubgraphRegistrar<
+            IpfsSubgraphAssignmentProvider<
+                SubgraphInstanceManager<graph_store_postgres::SubgraphStore>,
+            >,
+            graph_store_postgres::SubgraphStore,
+            PanicSubscriptionManager,
+        >,
+    > = Arc::new(IpfsSubgraphRegistrar::new(
+        &logger_factory,
+        link_resolver.cheap_clone(),
+        subgraph_provider.clone(),
+        subgraph_store.clone(),
+        panicking_subscription_manager,
+        blockchain_map.clone(),
+        node_id.clone(),
+        SubgraphVersionSwitchingMode::Instant,
+    ));
+
+    SubgraphRegistrar::create_subgraph(subgraph_registrar.as_ref(), subgraph_name.clone())
+        .await
+        .expect("unable to create subgraph");
+
+    SubgraphRegistrar::create_subgraph_version(
+        subgraph_registrar.as_ref(),
+        subgraph_name.clone(),
+        deployment.hash.clone(),
+        node_id.clone(),
+        None,
+        Some(BlockPtr {
+            hash: start_block.hash(),
+            number: start_block.number(),
+        }),
+    )
+    .await
+    .expect("failed to create subgraph version");
+
+    TestContext {
+        logger_factory,
+        provider: subgraph_provider,
+        store: subgraph_store,
+    }
+}
+
+struct StaticStreamBuilder<C: Blockchain> {
+    events: Vec<BlockStreamEvent<C>>,
+}
+
+#[async_trait]
+impl<C: Blockchain> BlockStreamBuilder<C> for StaticStreamBuilder<C> {
+    fn build_firehose(
+        &self,
+        _chain: &C,
+        _deployment: DeploymentLocator,
+        _block_cursor: Option<String>,
+        _start_blocks: Vec<graph::prelude::BlockNumber>,
+        _subgraph_current_block: Option<graph::blockchain::BlockPtr>,
```

Comment:
Here and in `build_polling`, we probably want to use the `block_cursor` or  the `subgraph_current_block` as a starting index into `self.events`, for tests that exercise block stream restarts. You don't need to think through this now, but it's worth adding a comment so we don't forget.